### PR TITLE
[OLD] Add yurthub podenvupdater filter

### PR DESCRIPTION
**Pull request by techworldhello**
/kind feature

#### What this PR does / why we need it:

Adds a filter in Yurthub that sets the `KUBERNETES_SERVICE_HOST` environment variable to the Yurthub proxy IP. This enhancement would ensure node autonomy for edge nodes when they are operating offline.

#### Which issue(s) this PR fixes:

Fixes https://github.com/openyurtio/openyurt/issues/2158

#### Special notes for your reviewer:

/assign @rambohe-ch 

#### Does this PR introduce a user-facing change?

```release-note

NONE

```

#### other Note

I noticed that documentation on new features are added to the changelog.md when a new version is created eg. https://github.com/openyurtio/openyurt/pull/1425/files. But if the maintainers prefer that this is added as a part of this PR please let me know,

### CODE REVIEW

File: pkg/yurthub/filter/podenvupdater/filter.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	default:
+		return v
+	}
+}
+
+func (pef *podEnvFilter) mutatePodEnv(req *corev1.Pod) *corev1.Pod {
+	proxyIP := pef.host
+	if proxyIP == "" {
+		proxyIP = yurthubHost
+	}
```
Comment (rambohe-ch) [inline pkg/yurthub/filter/podenvupdater/filter.go:68]: It's not needed to verify `pef.host` whether it is an empty string or not in Filter function. we can use `pef.host` directly without verification.

Comment (techworldhello) [reply pkg/yurthub/filter/podenvupdater/filter.go]: fair enough, I've removed the verification in a rebase

File: pkg/yurthub/filter/podenvupdater/filter.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package podenvupdater
```
Comment (rambohe-ch) [inline pkg/yurthub/filter/podenvupdater/filter.go:1]: please add OpenYurt license header

Comment (techworldhello) [reply pkg/yurthub/filter/podenvupdater/filter.go:17]: added in a rebase

File: pkg/yurthub/filter/podenvupdater/filter_test.go [hunk 3]
```diff
@@ -0,0 +1,246 @@
+package podenvupdater
```
[the remark this answers is not in the document]
Comment (techworldhello) [reply pkg/yurthub/filter/podenvupdater/filter_test.go:17]: added in a rebase

**Pull request closed without merging**